doc/website/blog/tmp-versioned-migrations-sum-file.md

```diff
@@ -0,0 +1,249 @@
+go generate ./...
+docker run --rm --name atlas-sum --detach --env MYSQL_ROOT_PASSWORD=pass --env MYSQL_DATABASE=ent -p 3306:3306 mysql
+```
+:::
+
+The first step is to tell the migration engine to create and manage the `atlas.sum` by using the `schema.WithSumFile()`
+option. The below example uses and [instantiated Ent client](/docs/versioned-migrations.md#from-client) to generate new
+migration files:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"ent-sum-file/ent"
+
+	"ariga.io/atlas/sql/migrate"
+	"entgo.io/ent/dialect/sql/schema"
+	_ "github.com/go-sql-driver/mysql"
+)
+
+func main() {
+	client, err := ent.Open("mysql", "root:pass@tcp(localhost:3306)/ent")
+	if err != nil {
+		log.Fatalf("failed connecting to mysql: %v", err)
+	}
+	defer client.Close()
+	ctx := context.Background()
+	// Create a local migration directory.
+	dir, err := migrate.NewLocalDir("migrations")
+	if err != nil {
+		log.Fatalf("failed creating atlas migration directory: %v", err)
+	}
+	// Write migration diff.
+	// highlight-start
+	err = client.Schema.NamedDiff(ctx, os.Args[1], schema.WithDir(dir), schema.WithSumFile())
+	// highlight-end
+	if err != nil {
+		log.Fatalf("failed creating schema resources: %v", err)
+	}
+}
+```
+
+After creating a migrations directory and running the above commands you should see `golang-migrate/migrate` compatible
+migration files and in addition to that the `atlas.sum` file with the following contents:
+
+```shell
+mkdir migrations
+go run -mod=mod main.go initial
+```
+
+```sql title="20220504114411_initial.up.sql"
+-- create "users" table
+CREATE TABLE `users` (`id` bigint NOT NULL AUTO_INCREMENT, PRIMARY KEY (`id`)) CHARSET utf8mb4 COLLATE utf8mb4_bin;
+
+```
+
+```sql title="20220504114411_initial.down.sql"
+-- reverse: create "users" table
+DROP TABLE `users`;
+
+```
+
+```text title="atlas.sum"
+h1:SxbWjP6gufiBpBjOVtFXgXy7q3pq1X11XYUxvT4ErxM=
+20220504114411_initial.down.sql h1:OllnelRaqecTrPbd2YpDbBEymCpY/l6ihbyd/tVDgeY=
+20220504114411_initial.up.sql h1:o/6yOczGSNYQLlvALEU9lK2/L6/ws65FrHJkEk/tjBk=
+
+```
+
+As you can see the `atlas.sum` file contains one entry for each migration file generated. With the `atlas.sum`
+generation file enabled, both Team A and Team B will have such a file once they generate migrations for a schema change.
+Now the version control will raise a merge conflict once the second Team attempts to merge their feature. 
+
+![atlas-versioned-migrations-no-conflict](https://entgo.io/images/assets/migrate/conflict.svg)
+
+:::note
+You need to have the Atlas CLI installed on your system in order to follow along the next steps in this post. If you
+haven't yet, please follow the installation
+instructions [here](https://atlasgo.io/cli/getting-started/setting-up#install-the-cli).
+:::
+
+You can check at any time, if your `atlas.sum` file is in sync with the migration directory with the following command (
+which should not output any errors now):
+
+```shell
+atlas migrate validate
+```
```

Comment:
Since this is a Go module example, I'll suggest to use `go run -mod=mod ariga.io/atlas/cmd/atlas` for now.

Reply:
Instead of installing the CLI? No bad idea.